crates/bevy_asset/src/asset_server.rs

```diff
@@ -377,7 +393,7 @@ impl AssetServer {
         for child_path in self.server.asset_io.read_directory(path.as_ref())? {
             if self.server.asset_io.is_directory(&child_path) {
                 handles.extend(self.load_folder(&child_path)?);
-            } else {
+            } else if f(child_path.as_path()) {
```

Comment:
On second thought I should probably move this to before the `is_directory` call...
This way users can also conditionally load sub-directories